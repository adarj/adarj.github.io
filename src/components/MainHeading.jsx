import React from 'react';
import styled from 'styled-components';
import tw from 'tailwind.macro';

const Title = styled.h1`${tw`text-white`};`;
const Subtitle = styled.h2`${tw`text-white`};`;
const Border = styled.div`${tw`border-solid border-white border-2 lg:p-32 lg:py-8 p-16 py-4`};`;

const MainHeading = () => (
  <Border>
    <Title data-cy="title">Akil Darjean</Title>
    <Subtitle data-cy="subtitle">Software Developer</Subtitle>
  </Border>
);

export default MainHeading;
