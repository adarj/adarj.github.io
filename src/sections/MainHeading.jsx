import React from 'react';
import Hero from 'react-lazy-hero';
import styled from 'styled-components';
import tw from 'tailwind.macro';

const Title = styled.h1`${tw`text-white m-0 mb-6 text-6xl font-normal`};`;
const Subtitle = styled.h2`${tw`text-white m-0 text-3xl`};`;
const Border = styled.div`${tw`border-solid border-white border-3 lg:p-32 lg:py-8 p-6 py-4`};`;

const MainHeading = (props) => {
  const title = "Akil Darjean";
  const subtitle = "Software Developer";

  return (
    <Hero imageSrc={props.image} color="#000" opacity={0.6} minHeight="100vh">
      <Border>
        <Title data-cy="title">{title}</Title>
        <Subtitle data-cy="subtitle">{subtitle}</Subtitle>
      </Border>
    </Hero>
  );
};

export default MainHeading;
