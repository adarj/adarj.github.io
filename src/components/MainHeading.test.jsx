import React from 'react';
import { shallow } from 'enzyme';

import MainHeading from './MainHeading';
import heroBg from '../images/hero-bg.jpg';

describe('Main Heading', () => {
  it('renders correctly', () => {
    const tree = shallow(<MainHeading image={heroBg} title="John Doe" subtitle="Lorem Ipsum" />);
    expect(tree).toMatchSnapshot();
  });
});

