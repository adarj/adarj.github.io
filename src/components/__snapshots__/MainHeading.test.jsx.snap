// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Heading renders correctly 1`] = `
<styled.div>
  <styled.h1
    data-cy="title"
  >
    Akil Darjean
  </styled.h1>
  <styled.h2
    data-cy="subtitle"
  >
    Software Developer
  </styled.h2>
</styled.div>
`;
